src/components/AddressSearch.js

```diff
@@ -23,134 +23,125 @@ const propTypes = {
-                },
-            }}
-            styles={{
-                textInputContainer: [styles.flexColumn],
-                listView: [
-                    !displayListViewBorder && styles.googleListView,
-                    displayListViewBorder && styles.borderTopRounded,
-                    displayListViewBorder && styles.borderBottomRounded,
-                    displayListViewBorder && styles.mt1,
-                    styles.overflowAuto,
-                    styles.borderLeft,
-                    styles.borderRight,
-                ],
-                row: [
-                    styles.pv4,
-                    styles.ph3,
-                    styles.overflowAuto,
-                ],
-                description: [styles.googleSearchText],
-                separator: [styles.googleSearchSeparator],
-            }}
-            onLayout={(event) => {
-                // We use the height of the element to determine if we should hide the border of the listView dropdown
-                // to prevent a lingering border when there are no address suggestions.
-                // The height of the empty element is 2px (1px height for each top and bottom borders)
-                setDisplayListViewBorder(event.nativeEvent.layout.height > 2);
-            }}
-        />
-    );
-};
+    }
+
+    render() {
+        return (
+            <GooglePlacesAutocomplete
+                fetchDetails
+                suppressDefaultStyles
+                enablePoweredByContainer={false}
+                onPress={(data, details) => {
+                    this.saveLocationDetails(details);
+
+                    // After we select an option, we set displayListViewBorder to false to prevent UI flickering
+                    this.setState({displayListViewBorder: false});
+                }}
+                query={{
+                    key: 'AIzaSyC4axhhXtpiS-WozJEsmlL3Kg3kXucbZus',
+                    language: this.props.preferredLocale,
+                    types: 'address',
+                    components: 'country:us',
+                }}
+                requestUrl={{
+                    useOnPlatform: 'web',
+                    url: `${CONFIG.EXPENSIFY.URL_EXPENSIFY_COM}api?command=Proxy_GooglePlaces&proxyUrl=`,
+                }}
+                textInputProps={{
+                    InputComp: ExpensiTextInput,
+                    label: this.props.label,
+                    containerStyles: this.props.containerStyles,
+                    errorText: this.props.errorText,
+                    value: this.props.value,
+                    onChangeText: (text) => {
+                        // Line of code https://github.com/FaridSafi/react-native-google-places-autocomplete/blob/47d7223dd48f85da97e80a0729a985bbbcee353f/GooglePlacesAutocomplete.js#L148
+                        // will call onChangeText passing '' after the component renders the first time, clearing its value. Why? who knows, but we have to skip it.
```

Comment:
Maybe just me, but I think we should either figure out why it happens or just say we are working around a feature of the library.

Reply:
But ideally we don't do this at all and there is no explanation necessary 😄

Reply:
Yeah, I guess it is not the best comment. The point is, I know why it happens (like what code in the library is creating the problem), but I don't know the reason why they decided to do that (doesn't make sense to me).

>we are working around a feature of the library.
This ^